src/integrationTest/java/org/opensearch/test/framework/cluster/TestRestClient.java

```diff
@@ -111,6 +112,13 @@ public HttpResponse get(String path, Header... headers) {
         return executeRequest(new HttpGet(getHttpServerUri() + "/" + path), headers);
     }
 
+    public HttpResponse getWithoutLeadingSlash(String path, Header... headers) {
+        URI uri = URI.create(getHttpServerUri());
+        uri = uri.resolve(path);
```

Comment:
Had to switch this because `setPath` is not available on `HttpUriRequest` in Apache HttpCore4, it was added in v5: https://github.com/apache/httpcomponents-core/blob/81a3c82473e93cdf8ef5b7873e44046f24c6d871/httpcore5/src/main/java/org/apache/hc/core5/http/HttpRequest.java#L58-L64